src/backend/tests/it/user_token.rs

```diff
@@ -0,0 +1,408 @@
+
+    let tokens: Vec<UserToken> = vec![MOCK_TOKEN.clone(), ANOTHER_TOKEN.clone()];
+
+    let result = update_call::<()>(&pic_setup, caller, "set_many_user_tokens", tokens.clone());
+
+    assert!(result.is_ok());
+
+    let add_token_results =
+        query_call::<Vec<UserToken>>(&pic_setup, caller, "list_user_tokens", ());
+
+    assert!(add_token_results.is_ok());
+
+    let expected_tokens: Vec<UserToken> = vec![
+        MOCK_TOKEN.clone_with_incremented_version(),
+        ANOTHER_TOKEN.clone_with_incremented_version(),
+    ];
+
+    assert_tokens_data_eq(&add_token_results.clone().unwrap(), &expected_tokens);
+
+    let update_token: UserToken = UserToken {
+        enabled: Some(false),
+        version: add_token_results.clone().unwrap().get(0).unwrap().version,
+        ..MOCK_TOKEN.clone()
+    };
+
+    let update_another_token: UserToken = UserToken {
+        enabled: Some(true),
+        version: add_token_results.clone().unwrap().get(1).unwrap().version,
+        ..ANOTHER_TOKEN.clone()
+    };
+
+    let update_tokens: Vec<UserToken> = vec![update_token.clone(), update_another_token.clone()];
+
+    let update_result = update_call::<()>(
+        &pic_setup,
+        caller,
+        "set_many_user_tokens",
+        update_tokens.clone(),
+    );
+
+    assert!(update_result.is_ok());
+
+    let results = query_call::<Vec<UserToken>>(&pic_setup, caller, "list_user_tokens", ());
+
+    let expected_tokens: Vec<UserToken> = vec![
+        update_token.clone_with_incremented_version(),
+        update_another_token.clone_with_incremented_version(),
+    ];
+
+    assert!(results.is_ok());
+
+    let updated_tokens = results.unwrap();
+
+    assert_tokens_data_eq(&updated_tokens, &expected_tokens);
+}
+
+#[test]
+fn test_disable_user_token() {
+    let pic_setup = setup();
+
+    let caller = Principal::from_text(CALLER.to_string()).unwrap();
+
+    let result = update_call::<()>(&pic_setup, caller, "set_user_token", MOCK_TOKEN.clone());
+
+    assert!(result.is_ok());
+
+    let add_token_result = query_call::<Vec<UserToken>>(&pic_setup, caller, "list_user_tokens", ());
+
+    assert!(add_token_result.is_ok());
+
+    let update_token: UserToken = UserToken {
+        enabled: Some(false),
+        version: add_token_result.unwrap().get(0).unwrap().version,
+        ..MOCK_TOKEN.clone()
+    };
+
+    let update_result =
+        update_call::<()>(&pic_setup, caller, "set_user_token", update_token.clone());
+
+    assert!(update_result.is_ok());
+
+    let results = query_call::<Vec<UserToken>>(&pic_setup, caller, "list_user_tokens", ());
+
+    let expected_tokens: Vec<UserToken> = vec![update_token.clone_with_incremented_version()];
+
+    assert!(results.is_ok());
+
+    let updated_tokens = results.unwrap();
+
+    assert_tokens_data_eq(&updated_tokens, &expected_tokens);
+}
+
+#[test]
+fn test_list_user_tokens() {
+    let pic_setup = setup();
+
+    let caller = Principal::from_text(CALLER.to_string()).unwrap();
+
```

Comment:
I would recommend listing user tokens before to demonstrate that the list was empty.

Reply:
Sure! 👉 https://github.com/dfinity/oisy-wallet/pull/1641